logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MarkMessagesAsNotifiedUseCase.kt

```diff
@@ -0,0 +1,14 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.functional.Either
+
+class MarkMessagesAsNotifiedUseCase(private val messageRepository: MessageRepository) {
+
+    suspend operator fun invoke(conversationId: ConversationId? = null): Either<CoreFailure, Unit> {
```

Comment:
Maybe we could return a Result type here that we can define here at this class level. If I'm not wrong, we are trying to move away from exposing `Either` to reloaded, also because generics could cause problems in other platforms like: iOS, web